src/Makefile

```diff
@@ -1358,6 +1356,44 @@ build_syscollector: build_shared_modules build_sysinfo
 	cd ${SYSCOLLECTOR} && mkdir -p build && cd build && cmake ${WIN_CMAKE_OPTS} ${SOLARIS_CMAKE_OPTS} ${SYSCOLLECTOR_TEST} ${SYSCOLLECTOR_RELEASE_TYPE} .. && ${MAKE}
 endif
 
+#### rootcheck #####
+
+rootcheck_c := $(wildcard rootcheck/*.c)
+rootcheck_o := $(rootcheck_c:.c=.o)
+rootcheck_o_lib := $(filter-out rootcheck/rootcheck-config.o, ${rootcheck_o})
+rootcheck_o_cmd := $(filter-out rootcheck/config.o, ${rootcheck_o})
+
+
+rootcheck/%.o: rootcheck/%.c
+	${OSSEC_CC} ${OSSEC_CFLAGS} -DARGV0=\"rootcheck\" -c $^ -o $@
+
+rootcheck.a: ${rootcheck_o_lib}
+	${OSSEC_LINK} $@ $^
+	${OSSEC_RANLIB} $@
+
+
+#### syscheck ######
+
+syscheck_sql := syscheckd/db/schema_fim_db.sql
+
+syscheck_c := $(wildcard syscheckd/*.c)
+syscheck_c += $(wildcard syscheckd/db/*.c)
+syscheck_c += $(wildcard syscheckd/whodata/*.c)
+syscheck_c += $(wildcard syscheckd/registry/*.c)
+
+syschecklib_o := $(syschecklib_c:.c=.o) $(syscheck_sql:.sql=.o)
+syscheck_o := $(syschecklib_o:.c=.o) $(syscheck_c:.c=.o) $(syscheck_sql:.sql=.o)
+syscheck_eventchannel_o := $(syscheck_c:.c=-event.o) $(syscheck_sql:.sql=.o)
+
+syscheckd/db/schema_fim_db.o: syscheckd/db/schema_fim_db.sql
+	${QUIET_CC}echo 'const char *schema_fim_sql = "'"`cat $< | tr -d \"\n\"`"'";' | ${MING_BASE}${CC} ${OSSEC_CFLAGS} -xc -c -o $@ -
+
+syscheckd/%.o: syscheckd/%.c
+	${OSSEC_CC} ${OSSEC_CFLAGS} -DARGV0=\"wazuh-modulesd:syscheck\" -c $^ -o $@
```

Comment:
Considering how `logcollector` was done and how `execd` will be done I'd recommend unify the way of how log tag is defined. Maybe it will be more clear if you just leave here (in Makefile):  `... -DARGV0=\"wazuh-modulesd\" ...` and, in `src/syscheckd/syscheck.h` add the following:

```
#ifndef ARGV0
#define ARGV0 "wazuh-modulesd"
#endif

#define WM_SYSCHECK_LOGTAG ARGV0 ":syscheck" // Tag for log messages
```
and, of course, change all mtinfo, mterror, twarn, etc from `ARGV0` to `WM_SYSCHECK_LOGTAG`

That way, you will not need all the unit tests defines `(#define SYSCHECK_MODULE_NAME "wazuh-modulesd:syscheck")`